src/server/http/__tests__/base_path_rewrite.js

```diff
@@ -0,0 +1,188 @@
+import { Server } from 'hapi';
+import expect from 'expect.js';
+import sinon from 'sinon';
+
+import { setupBasePathRewrite } from '../setup_base_path_rewrite';
+
+describe('server / base path rewriting', () => {
+  function createServer({ basePath, rewriteBasePath }) {
+    const config = {
+      get: sinon.stub()
+    };
+
+    config.get.withArgs('server.basePath')
+      .returns(basePath);
+    config.get.withArgs('server.rewriteBasePath')
+      .returns(rewriteBasePath);
+
+    const server = new Server();
+    server.connection({ port: 0 });
+    setupBasePathRewrite({}, server, config);
+
+    server.route({
+      method: 'GET',
+      path: '/',
+      handler(req, reply) {
+        reply('resp:/');
+      }
+    });
+
+    server.route({
+      method: 'GET',
+      path: '/foo',
+      handler(req, reply) {
+        reply('resp:/foo');
+      }
+    });
+
+    return server;
+  }
+
+  describe('no base path', () => {
+    it('/bar => 404', async () => {
+      const server = createServer({ basePath: '', rewriteBasePath: false });
```

Comment:
__nit:__ maybe move `const server = createServer({ basePath: '', rewriteBasePath: false });` to `beforeEach` (for this suite and suites below)?